src/Functions/toRelativeYearNum.cpp

```diff
@@ -11,7 +11,41 @@ using FunctionToRelativeYearNum = FunctionDateOrDateTimeToSomething<DataTypeUInt
 
 REGISTER_FUNCTION(ToRelativeYearNum)
 {
-    factory.registerFunction<FunctionToRelativeYearNum>();
+    FunctionDocumentation::Description description_toRelativeYearNum = R"(
```

Comment:
Re variable names: Can we not just use `description`, `syntax` etc., instead of `description_toRelativeYearNum`, `syntax_toRelativeYearNum` etc.? This file contains just one function definition, there is no ambiguity.